company/strategy.md

```diff
@@ -1,36 +1,129 @@
+
+### 3-year vision {#3-year-vision}
+
+> Problem statement: Developers spend too much time on repetitive and overhead tasks, not on writing code to solve unique business problems.
+
+Sourcegraph understands higher-level software development patterns and manages processes in the background for you, which lets you spend more time on the higher-leverage tasks that truly require your human brain.
+
+- Sourcegraph has support for codifying concepts such as code ownership, tests, deployment, logging, tracing, performance, monitoring, build processes, and APIs. How you implement these is up to you. (If you do something non-standard, you just need to teach Sourcegraph, for example, how to determine where a particular file's code is deployed.)
+- Sourcegraph has purpose-built new workflows that build on these concepts for common tasks. For example, if an application is reported down by your monitoring service, the first step is to visit a Sourcegraph page that shows all relevant information (recent deployments, recent commits, aberrant traces and logs, code owners, etc.) and presents your organization's preferred actions (roll back, create war room chat channel, etc.).
+- When you make a change to an API, you can include "metacode" that knows how to update all usages of the changed API. Sourcegraph will show you a compatibility report and propose those changes on all downstream callers along with your change.
+- You can discover and more easily use libraries, with statistics about usage patterns and a code marketplace.
+
+> Unlike the [1-year vision](#1-year-vision), our 3-year vision involves Sourcegraph modifying existing workflows and being proactive.
+
+### 10-year vision {#10-year-vision}
+
+> Problem statement: Software development is a black box to everyone else in an organization. Communication and collaboration in both directions is poor.
+
+Everyone involved in the software value chain (developers, sales, product, marketing, support, customers/users, etc.) uses Sourcegraph to coordinate work and share information.
+
+From a developer's point of view, they can go from the code for a particular feature to a list of all customers using it, all feedback and support requests about it, and all upcoming features that relate to it.
+
+From a customer's and salesperson's point of view, they can see all in-progress or planned features for the particular customer. For any given feature, they can see the current state of development.
+
+> Unlike the [3-year vision](#3-year-vision), our 10-year vision involves more kinds of people using Sourcegraph (not just people who write code directly).
+
+## Principles
+
+- Sourcegraph is universal code search, not universal "everything" search. Any additional data types in our search need to be relevant to the software development workflow.
+- We're a mass-market company and must maintain broad appeal. We want every developer, not just a specific niche audience, to use Sourcegraph.
```

Comment:
This strikes me as more of a desired outcome, rather than a principle. "Mass-market" also sounds kinda negative (mass-market == commodity == cheap == primarily concerned about cash).

In my view, a more principle-y statement would be: "We wish to serve developers across a broad spectrum of communities, styles, and backgrounds. We are un-opinionated when it comes to specific languages and technologies, as we recognize that innovation comes from across the spectrum. We trust our customers to choose the best technologies for the job, and seek to make them productive in whatever stack they choose."

Reply:
Updated to just say:

```
- We want every developer, not just a specific niche audience, to use Sourcegraph.
  - We won't adopt polarizing public stances that would divide our audience.
```

Reply:
>  - We won't adopt polarizing public stances that would divide our audience.

If this is intended to refer to technical stances (languages, frameworks, etc.)? Just from an outsider's POV, it sounds more broad than that the way it's phrased.

Reply:
I intended it to be about technical stances, but it would also apply to other stances. Of course, it's a judgment call as to whether a particular stance would "divide" our audience or appeal to it.